packages/dashboard/src/components/delivery-alert-store.tsx

```diff
@@ -0,0 +1,539 @@
+        if (resolvedTaskIds.includes(taskId)) {
+          continue;
+        }
+        const alertData = taskIdToAlertsMap[taskId];
+        filteredAlertsMap[alertData.deliveryAlert.id] = alertData;
+      }
+      setAlerts(filteredAlertsMap);
+    })();
+
+    const sub = rmf.deliveryAlertObsStore.subscribe(async (deliveryAlert) => {
+      let state: TaskState | undefined = undefined;
+      if (deliveryAlert.task_id) {
+        try {
+          state = (await rmf.tasksApi.getTaskStateTasksTaskIdStateGet(deliveryAlert.task_id)).data;
+        } catch {
+          console.error(`Failed to fetch task state for ${deliveryAlert.task_id}`);
+        }
+      }
+      filterAndPushDeliveryAlert(deliveryAlert, state);
+    });
+    return () => sub.unsubscribe();
+  }, [rmf]);
+
+  const onCancel = React.useCallback<Required<DeliveryWarningDialogProps>['onCancel']>(
+    async (delivery_alert_id, task_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('tasks and delivery alert api not available');
+        }
+        await rmf.tasksApi?.postCancelTaskTasksCancelTaskPost({
+          type: 'cancel_task_request',
+          task_id: task_id,
+        });
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
+          delivery_alert_id,
+          'cancelled',
+        );
+        appController.showAlert('success', 'Successfully cancelled task');
+      } catch (e) {
+        appController.showAlert('error', `Failed to cancel task: ${(e as Error).message}`);
+      }
+    },
+    [rmf, appController],
+  );
+
+  const onOverride = React.useCallback<Required<DeliveryWarningDialogProps>['onOverride']>(
+    async (delivery_alert_id, task_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('delivery alert api not available');
+        }
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
+          delivery_alert_id,
+          'override',
+        );
+        const taskReferenceText = task_id ? `, continuing with task ${task_id}` : '';
+        appController.showAlert(
+          'success',
+          `Overriding delivery alert ${delivery_alert_id}${taskReferenceText}`,
+        );
+      } catch (e) {
+        const taskReferenceText = task_id ? ` and continue with task ${task_id}` : '';
+        appController.showAlert(
+          'error',
+          `Failed to override delivery alert ${delivery_alert_id}${taskReferenceText}: ${
+            (e as Error).message
+          }`,
+        );
+      }
+    },
+    [rmf, appController],
+  );
+
+  const onResume = React.useCallback<Required<DeliveryWarningDialogProps>['onResume']>(
+    async (delivery_alert_id, task_id) => {
+      try {
+        if (!rmf) {
+          throw new Error('delivery alert api not available');
+        }
+        await rmf.deliveryAlertsApi?.updateDeliveryAlertActionDeliveryAlertsDeliveryAlertIdActionPost(
```

Comment:
Maybe the '?' is not necessary

Reply:
https://github.com/open-rmf/rmf-web/pull/768/commits/83666bba825ec114cc679179300587c135db958a